### PR TITLE
Add correct endpoint and serialization for submission creation and test runs

### PR DESCRIPTION
**Pull request by iHiD**
@kntsoriano I've put this together so you can integrate against it.

- **Commit 1:** Removes the `major` flag from submissions (as it's now replaced by iterations/submissions being seperate). It also renames `pending` to `queued` in the various submission statues.
- **Commit 2:** Adds the correct endpoint for the editor to hit: `POST /api/solution/:solution_uuid/submissions`
- **Commit 3:** Adds a serializer for submissions
- **Commit 4:** Change the serializer for test runs. I think that we should be subscribing to the submission (via the UUID) and then receive any test runs that are processed on this channel.

I've decided not to serialize the test run as part of (3), because I think it should be coming seperately via the `Submission::TestRunsChannel` but I could be persuaded to have it on there as well if it's useful elsewhere.

### CODE REVIEW

**Review by kntsoriano — APPROVED** [review]
Looks good!

File: app/commands/submission/create.rb [hunk 1]
```diff
@@ -70,35 +71,25 @@ def init_services
...
-      end
+      # TODO: Move to iteration create
+      # jobs += [
+      #   Thread.new do
+      #     Submission::Analysis::Init.(submission_uuid, solution.track.slug, solution.exercise.slug, s3_uri)
+      #   end,
+      #   Thread.new do
+      #     Submission::Representation::Init.(submission_uuid, solution.track.slug, solution.exercise.slug, s3_uri)
+      #   end
+      # ]
```
Comment (ErikSchierboom) [inline app/commands/submission/create.rb:82]: This means temporarily there won't be any analyses or representations being run?

File: test/controllers/api/submissions_controller_test.rb [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  end
+
+  test "create should 404 if the solution belongs to someone else" do
+    setup_user
+    solution = create :concept_solution
+    post api_solution_submissions_path(solution.uuid), headers: @headers, as: :json
+    assert_response 403
+    expected = { error: {
+      type: "solution_not_accessible",
+      message: I18n.t('api.errors.solution_not_accessible')
```
Comment (ErikSchierboom) [inline test/controllers/api/submissions_controller_test.rb:25]: This looks like an implementation detail. Maybe just have the error message itself here? It does make the test a bit more brittle, but it doesn't tie the test into any implementation detail.

**Review by ErikSchierboom — APPROVED** [review]

File: app/commands/submission/create.rb (hunk 1 above)
Comment (iHiD) [reply app/commands/submission/create.rb:82]: Correct. But only for a day or two.

File: test/controllers/api/submissions_controller_test.rb (hunk 2 above)
Comment (iHiD) [reply test/controllers/api/submissions_controller_test.rb:25]: Yeah - that's an interesting point. So right now we do this for all the API tests, so I'll keep this to be consistent, but we should consider changing it everywhere. There's pros and cons to it.

Comment (ErikSchierboom) [reply test/controllers/api/submissions_controller_test.rb:25]: Absolutely

**Pull request merged**